code/datums/components/temporary_mute.dm

```diff
@@ -0,0 +1,45 @@
+/datum/component/temporary_mute
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// A message to tell the user when they attempt to speak, if any
+	var/on_speak_message = ""
+	/// A message to tell the user when they become no longer mute, if any
+	var/on_unmute_message = ""
+	/// How long after the component's initialization it should be deleted. -1 means it will never delete
+	var/time_until_unmute = 5 MINUTES
+
+/datum/component/temporary_mute/Initialize(on_speak_message = "", on_unmute_message = "", time_until_unmute = 5 MINUTES)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.on_speak_message = on_speak_message
+	src.on_unmute_message = on_unmute_message
+	src.time_until_unmute = time_until_unmute
+	if(time_until_unmute != -1)
+		QDEL_IN(src, time_until_unmute)
+
+
+/datum/component/temporary_mute/RegisterWithParent()
+	..()
+	RegisterSignal(parent, COMSIG_LIVING_SPEAK, PROC_REF(on_speak))
+
+/datum/component/temporary_mute/UnregisterFromParent()
+	..()
+	if(parent)
+		to_chat(parent, SPAN_NOTICE(on_unmute_message))
+		UnregisterSignal(parent, COMSIG_LIVING_SPEAK)
+
+/datum/component/temporary_mute/proc/on_speak(
```

Comment:
This doesn't currently handle hivemind speaking for the lesser drone. `/mob/living/carbon/proc/hivemind_broadcast`

(Facehuggers aren't allowed to hivespeak so `/mob/living/carbon/xenomorph/say` sets `forced` to 1 which forces it to a say instead.

Reply:
The signal check probably just needs to be earlier in the say chain; but need to make sure all calls to say are then from that higher proc call.

Reply:
> This doesn't currently handle hivemind speaking for the lesser drone. `/mob/living/carbon/proc/hivemind_broadcast`
> 
> Facehuggers aren't allowed to hivespeak so `/mob/living/carbon/xenomorph/say` sets `forced` to 1 which forces it to a say instead.

this is what i get for assuming that testing hivemind mute w/ only facehugger would be enough